test/functional/wallet_multiwallet.py

```diff
@@ -129,14 +126,27 @@ def wallet_file(name):
         for wallet_name in to_load:
             self.nodes[0].loadwallet(wallet_name)
 
-        os.mkdir(wallet_dir('no_access'))
-        os.chmod(wallet_dir('no_access'), 0)
-        try:
-            with self.nodes[0].assert_debug_log(expected_msgs=['Error scanning']):
-                walletlist = self.nodes[0].listwalletdir()['wallets']
-        finally:
-            # Need to ensure access is restored for cleanup
-            os.chmod(wallet_dir('no_access'), stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+        # Tests for possible scanning errors:
+        # 0. Baseline, no errors.
+        with self.nodes[0].assert_debug_log(expected_msgs=[], unexpected_msgs=['Error scanning']):
+            walletlist = self.nodes[0].listwalletdir()['wallets']
+        assert_equal(sorted(map(lambda w: w['name'], walletlist)), sorted(in_wallet_dir))
+        # 1. "Permission denied" error.
+        if platform.system() != 'Windows' and os.geteuid() != 0:
+            os.mkdir(wallet_dir('no_access'))
+            os.chmod(wallet_dir('no_access'), 0)
+            try:
+                with self.nodes[0].assert_debug_log(expected_msgs=['Error scanning']):
+                    walletlist = self.nodes[0].listwalletdir()['wallets']
+            finally:
+                # Need to ensure access is restored for cleanup
+                os.chmod(wallet_dir('no_access'), stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+            assert_equal(sorted(map(lambda w: w['name'], walletlist)), sorted(in_wallet_dir))
```

Comment:
Making this check conditional on platform is undocumented in the commit message, and seems orthogonal to the goal of testing errors individually. I think it would be best to split this out in a separate commit and document it?

Reply:
I don't think these changes can be meaningfully separated. Applying either change independently would break the test or render it incomplete.